docs/architecture.md

```diff
@@ -105,6 +116,15 @@ It is also worth noting that `NightshadeRuntime` is the struct that implements `
 
 ## Cross Cutting Concerns
 
+## State update
```

Comment:
This section is misplaced, `### Logging & Observability` is intended to be under `## Cross Cutting Concerns`

Reply:
Ah sorry I actually intend to have `### State update`, unless I misunderstand what you mean by `Cross Cutting Concerns`

Reply:
Yeah, I am not sure I'd call this a cross-cutting concern. It seems that this is in chain's purview, as it calls both the catchup and apply?

That is, in my mind cross-cutting concerns are for "aspect-oriented" things which touch otherwise unrelated sections of code:

* what common logging framework do we use?
* what is the standard testing approach?
* how do we monitor performance?
* what's the common error handling strategy (speaking of which, documenting deterministic vs non-deterministic vs coordinated failures here would be a helpful)

Reply:
> It seems that this is in chain's purview

Not really. It can be updated in both runtime and chain. Where do you think is the best place to put it?

Reply:
>Not really. It can be updated in both runtime and chain. 

But it seems that the "caller" is, in both cases, the `chain`? Ie, it's the chain which calls `Runtime::apply` (indirectly, via `RuntimeAdapter::apply_transactions`). 

In the overview, I'd expect pointers to the call-sites: 

> two code paths may trigger state update. Normal update (processing of a new block) is initiated by chain in `???` and ultimately calls `Runtime::apply`. If a node is lagging behind, the chain engages the `catchup` routine.

Reply:
> But it seems that the "caller" is, in both cases, the chain? Ie, it's the chain which calls Runtime::apply (indirectly, via RuntimeAdapter::apply_transactions).

Yeah good point